### PR TITLE
Support tuple type also for memoryview shape arg

### PR DESCRIPTION
**Pull request by 24seconds**
## Description

This pr is to fix [test_base64.py](https://github.com/24seconds/RustPython/blob/dd773b3413017955db8639968b0a286adfa05e61/Lib/test/test_base64.py) test failure. Turned out most of them are related to `memoryview`'s `cast` function.

In the Cpython,  `cast` function's argument `shape` seems to support `list` and `tuple` type.
https://github.com/python/cpython/blob/af9ee57b96cb872df6574e36027cc753417605f9/Objects/memoryobject.c#L1367-L1418  

But RustPython currently supports `list` only. 
So this pr did those below
- Change `CastArgs` shape field type to `OptionalArg<PyObjectRef>`
- Check shape type is one of list or tuple
- convert shape properly

Because many `test_base64.py` tests are related to this problem, most of them are resolved after fix.

For the check, I ran these. If there is another thing that I should do, please give me guide :bow: 
```
cargo check
cargo fmt
cargo clippy
```

### CODE REVIEW

File: vm/src/builtins/memory.rs [hunk 1]
```diff
@@ -865,7 +876,7 @@ struct CastArgs {
     #[pyarg(any)]
     format: PyStrRef,
     #[pyarg(any, optional)]
-    shape: OptionalArg<PyListRef>,
+    shape: OptionalArg<PyObjectRef>,
```
Comment (youknowone) [inline vm/src/builtins/memory.rs:879]: If they take only `list` and `tuple`, `Either` works.
```suggestion
    shape: OptionalArg<Either<PyTupleRef, PyListRef>>,
```

If it takes any iterable, `ArgIterable` will fit.

**Review by youknowone — COMMENTED** [review]
We usually use types to narrow down to specific types when it is possible. Could you update it using `Either`?

File: vm/src/builtins/memory.rs (hunk 1 above)
Comment (24seconds) [reply vm/src/builtins/memory.rs:879]: Wow good. Let me try `Either`

**Review by 24seconds — COMMENTED** [review]
65d9e2f, NICE!

**Review by youknowone — APPROVED** [review]
Nice work, thank you!

**Pull request merged**